pennylane/measurements/mid_measure.py

```diff
@@ -218,6 +232,18 @@ def _merge(self, other: "MeasurementValue"):
         merged_measurement_ids = list(set(self.measurement_ids).union(set(other.measurement_ids)))
         merged_measurement_ids.sort()
 
+        merged_measurements = []
+        for item in self.measurements + other.measurements:
+            is_duplicate = False
+            for unique_item in merged_measurements:
+                if item is unique_item:
+                    is_duplicate = True
```

Comment:
is there no simpler way to do this? I guess `set()` will only do the first part, but I feel like there is something that exists to remove "true" duplicates.

Reply:
I tried looking up a more clean way of doing this but I could not find anything better than this

Reply:
Just had a revelation, `set` can be used.